src/main/kotlin/graphics/scenery/Arrow.kt

```diff
@@ -0,0 +1,168 @@
+ * to the former one; altogehter 7 segments drawn sequentially.
+ * The arrow head scales with the length of the arrow.
+ *
+ * @author Vladimir Ulman <ulman@mpi-cbg.de>
+ */
+class Arrow(var vector: GLVector) : Node("Arrow"), HasGeometry {
+    /** Size of one vertex (e.g. 3 in 3D) */
+    override val vertexSize: Int = 3
+    /** Size of one texcoord (e.g. 2 in 3D) */
+    override val texcoordSize: Int = 2
+    /** Geometry type -- Default for Line is [GeometryType.LINE] */
+    override var geometryType: GeometryType = GeometryType.LINE_STRIP_ADJACENCY
+    /** Vertex buffer */
+    override var vertices: FloatBuffer = BufferUtils.allocateFloat(30)
+    /** Normal buffer */
+    override var normals: FloatBuffer = BufferUtils.allocateFloat(30)
+    /** Texcoord buffer */
+    override var texcoords: FloatBuffer = BufferUtils.allocateFloat(20)
+    /** Index buffer */
+    override var indices: IntBuffer = IntBuffer.wrap(intArrayOf())
+
+    /** Shader property for the line's starting segment color. Consumed by the renderer. */
+    @ShaderProperty
+    var startColor = GLVector(0.0f, 1.0f, 0.0f, 1.0f)
+
+    /** Shader property for the line's color. Consumed by the renderer. */
+    @ShaderProperty
+    var lineColor = GLVector(1.0f, 1.0f, 1.0f, 1.0f)
+
+    /** Shader property for the line's end segment color. Consumed by the renderer. */
+    @ShaderProperty
+    var endColor = GLVector(0.7f, 0.5f, 0.5f, 1.0f)
+
+    /** Shader property for the line's cap length (start and end caps). Consumed by the renderer. */
+    @ShaderProperty
+    var capLength = 1
+
+    /** Shader property to keep track of the current number of vertices. Consumed by the renderer. */
+    @ShaderProperty
+    var vertexCount: Int = 0
+        private set
+
+    /** Shader property for the line's edge width. Consumed by the renderer. */
+    @ShaderProperty
+    var edgeWidth = 2.0f
+
+    init {
+		material = ShaderMaterial.fromClass(this::class.java, listOf(ShaderType.VertexShader, ShaderType.GeometryShader, ShaderType.FragmentShader))
+		material.cullingMode = Material.CullingMode.None
+
+	 	reshape(vector)
+    }
+
+    /**
+     * Changes the shape of this arrow.
+	  * TODO: make this method public
+     *
+     * @param p     The vector defining the shape of the arrow
+     */
+    fun reshape(vector: GLVector) {
+		//init the data structures
+	 	clearPoints()
+
+		/** create the vector shape */
+		//shortcut to null vector...
+		var zeroGLvec = GLVector(0.0f,0.0f,0.0f)
+
+		//first of the two mandatory surrounding fake points that are never displayed
+		addPoint(zeroGLvec)
+
+		//the main "vertical" segment of the vector
+		addPoint(zeroGLvec)
+		addPoint(vector)
+
+		//the first triangle:
+		//the shape of the triangle
+		var V = 0.1f * vector.magnitude()
+
+		//vector base is perpendicular to the input vector
+		var base = GLVector(-vector.y(), vector.x(), 0.0f)
+		var baseLen = base.magnitude()
+
+		if (baseLen == 0.f) {
```

Comment:
This is invalid syntax, as the CI complained :( Either 0.0f or 0f.

Reply:
changed.